rfcs/2023-12-25-proxy-implementation.md

```diff
@@ -0,0 +1,108 @@
+# SMP proxy implementation concerns
+
+* SMP proxy traffic should be indistinguishable from ordinary relay operation.
+* Proxy servers should be implemented by the same binary to prevent maintenance bloat.
+
+This leads to the proxy functionality added directly to the SMP protocol, raising its version.
+
+## Involved parties
+
+### Client
+
+> Client, from the servers' point of view, means the agent running inside a client app. An app/agent protocol is unchanged.
+
+A client that wants to utilize proxy functionality should request protocol version no lower than adds support for it.
```

Comment:
unclear. I think it's the other way around, app should not use proxy if server does not support it, and, optionally, refuse connection if above requirement cannot be met - controlled via NetworkConfig, similarly to how onion host options are controlled.